### PR TITLE
make loss.backward() work

### PR DESCRIPTION
**Pull request by allenhung1025**
I was trying to use gumbel_intersection in my training code. However, when calling the loss.backward(), there would be an error message. [Imgur](https://i.imgur.com/0AD09W6.png) which shows that there should not be an in-place operation with tunable parameter. After I made some changes in the intersection function, the code worked out.

### CODE REVIEW

**Comment by dhruvdcoder** [comment]
Thanks for creating the PR. I am not entirely sure if this issue is coming up due to the box-embeddings library—we have been able to backprop successfully while using gumbel intersection. Would you be able to create a small test case that isolates this issue this issue?
